File: src/main/java/no/novari/flyt/authorization/client/sourceapplications/model/HMSRegSourceApplication.java
package no.novari.flyt.authorization.client.sourceapplications.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class HMSRegSourceApplication extends BaseSourceApplication {
    public HMSRegSourceApplication(@Value("${fint.flyt.hmsreg.sso.client-id:#{null}}") String clientId) {
        super(6L, "HMSReg", clientId);
    }
}

File: src/main/kotlin/no/novari/flyt/authorization/client/sourceapplications/model/HMSRegSourceApplication.kt
package no.novari.flyt.authorization.client.sourceapplications.model

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class HMSRegSourceApplication(
    @Value("\${fint.flyt.hmsreg.sso.client-id:#{null}}")
    clientId: String?,
    @Value("\${fint.flyt.hmsreg.available:true}")
    available: Boolean,
) : BaseSourceApplication(
        id = 6L,
        displayName = "HMSReg",
        clientId = clientId,
        available = available,
    )
